### PR TITLE
Add support for ALTER TABLE statement in MongoDB

### PR DESCRIPTION
**Pull request by NaveenNitturu**
## Motivation and Context

[#23265 ](https://github.com/prestodb/presto/issues/23265)
Adding the ALTER TABLE support for MongoDB connector.

## Impact

Introducing a new feature in the MongoDB connector: support for ALTER TABLE statements to ADD, DROP and RENAME columns.

## Test Plan

Configured the MongoDB connector, by creating a catalog properties file in etc/catalog named mongodb.properties, to mount the MongoDb connector as the MongoDB catalog. Tested all three statements: ADD, DROP, and RENAME.

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

MongoDB Connector Changes
* Add support for MongoDB ``ALTER TABLE`` statement. :pr:`23266`
```

### CODE REVIEW

File: presto-mongodb/src/test/java/com/facebook/presto/mongodb/TestMongoIntegrationSmokeTest.java [hunk 1]
```diff
@@ -270,4 +270,18 @@ public void testRenameTable()
...
+    public void testAlterTablle()
+    {
+        assertUpdate("CREATE TABLE test_alter.tmp_alter_table (value bigint)");
+        MongoCollection<Document> collection = mongoQueryRunner.getMongoClient().getDatabase("test_alter").getCollection("tmp_alter_table");
+        collection.insertOne(new Document(ImmutableMap.of("value", 1)));
+
+        assertUpdate("ALTER TABLE test_alter.tmp_alter_table ADD COLUMN email varchar");
+        collection.insertOne(new Document(ImmutableMap.of("value", 2, "email", "example@example.com")));
+        assertQuery("SELECT email from test_alter.tmp_alter_table WHERE email IS NOT NULL", "SELECT 'example@example.com'");
+        assertUpdate("ALTER TABLE test_alter.tmp_alter_table RENAME COLUMN email TO email_id");
```
Comment (tdcmeehan) [inline presto-mongodb/src/test/java/com/facebook/presto/mongodb/TestMongoIntegrationSmokeTest.java:284]: Please query the renamed column

**Review by tdcmeehan — CHANGES_REQUESTED** [review]
Please squash commits

**Review by steveburnett — DISMISSED** [review]
LGTM! (docs) 

Pull branch, local doc build, looks good. Thanks!

**Comment by steveburnett** [comment]
Nits of suggested changes for the release note entry:

```
== RELEASE NOTES ==

MongoDB Connector Changes
* Add support for MongoDB ``ALTER TABLE`` statement. :pr:`23266`
```

File: presto-mongodb/src/test/java/com/facebook/presto/mongodb/TestMongoIntegrationSmokeTest.java (hunk 1 above)
Comment (NaveenNitturu) [reply presto-mongodb/src/test/java/com/facebook/presto/mongodb/TestMongoIntegrationSmokeTest.java:297]: added the query for renamed column.

**Comment by NaveenNitturu** [comment]
> Please squash commits

squashed the commits, Could you please review.

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**